### PR TITLE
Add Cloudtrail terraform scripts

### PR DESCRIPTION
**Pull request by jasonBirchall**
connects to https://github.com/ministryofjustice/cloud-platform/issues/307

**WHAT**
This PR contains the Terraform file that includes the creation of a Cloudtrail instance with an S3 bucket for log storage. All environments have been included and I've applied a lifecycle rule with relevant tagging. 

**WHY**
This connects to issue https://github.com/ministryofjustice/cloud-platform/issues/307. It will be applied to the Integration account. This will enable Cloudtrail for all regions and host in an S3 bucket.

### CODE REVIEW

File: Cloudtrail/main.tf [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    prefix                                 = ""
+    abort_incomplete_multipart_upload_days = 7
+    enabled                                = true
+
+    transition {
+      days          = 30
+      storage_class = "STANDARD_IA"
+    }
+
+    transition {
```
Comment (alkar) [inline Cloudtrail/main.tf]: Let's not go into glacier. It's then painful to retrieve anything. I doubt there's any real cost savings.

File: Cloudtrail/main.tf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailAclCheck",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::cloud-platform-cloudtrail-moj"
```
Comment (alkar) [inline Cloudtrail/main.tf]: Can we use interpolation here?

File: Cloudtrail/main.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        },
+        {
+            "Sid": "AWSCloudTrailWrite",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::cloud-platform-cloudtrail-moj/*",
```
Comment (alkar) [inline Cloudtrail/main.tf]: Same as above.

File: Cloudtrail/main.tf (hunk 2 above)
Comment (jasonBirchall) [reply Cloudtrail/main.tf]: Agreed, added a var for the bucket name and interpolated into the policy under cb8016a.

File: Cloudtrail/main.tf (hunk 1 above)
Comment (jasonBirchall) [reply Cloudtrail/main.tf]: After a chat, this has been agreed and pushed into cb8016a.

File: Cloudtrail/main.tf (hunk 3 above)
Comment (jasonBirchall) [reply Cloudtrail/main.tf]: Agreed, added a var for the bucket name and interpolated into the policy under cb8016a.

**Pull request merged**